simplegrid/cow.py

```diff
@@ -118,23 +110,24 @@ def step(self, observation):
 
         # Initial value of a is to favor one direction a little more than others
         # The random is to prevent creature from ending up in an oscillator
-        possible_actions = {a: (a + 2 * random.random()) / 100 for a in list(Action)[1:-1]}
+        possible_actions = {a: (a + 2 * random.random()) / 100 for a in list(Action)[:-1]}
 
         for col in range(size):
             x = col - view_distance
             for row in range(size):
+                reward = observation[col, row, 0]
                 y = row - view_distance
                 dist = abs(x) + abs(y)
-                if 0 < dist <= view_distance:
-                    value = observation[col, row]
-                    if value == MapFeature.GRASS.index:
-                        reward = 1
-                    elif dist == 1 and value == MapFeature.ROCK.index:
-                        reward = -1
+                if dist == 0:
+                    possible_actions[Action.GRAZE] += reward * 5
```

Comment:
Happy to add a comment, but we don't comment any of the other **weights** here. We divide by the distance, but that would lead to a division by zero :) so multiply by 5!